qacode/core/webs/pages/PageBase.py

```diff
@@ -40,6 +40,8 @@ def __init__(self,bot, url, by=By.CSS_SELECTOR, selectors=[], go_url=True):
         #more logic
         if go_url:
             self.go_page_url()
+        if len(selectors) > 0:
+            self.get_elements()
 
     def get_elements(self, selectors=[]):
```

Comment:
Normally is not expected for a method named `get_<whatever>s` to change attributes of the instance but to return a list/tuple or `<whatever>s`.

I think a good change would be one of:
a) rename the method to be `populate_elements`.
b) make the method to return the list of elements and assign them to the instance with something like `self.elements = self.get_elements(selectors)`.

Personally I prefer the second option but it's up to you ;)

Reply:
🥇